File: src/lib/runners/custom/cpp/first-bad-version.cpp
#include <iostream>
#include <vector>
#include <string>
#include <queue>
#include <sstream>
#include <climits>
#include <unordered_map>
#include <unordered_set>
#include <set>
#include <map>
#include <algorithm>
#include <numeric>
#include <cmath>
#include <stack>

using namespace std;

// --- Manual Parsers ---
string trim(const string& s) {
    size_t start = s.find_first_not_of(" \t\r\n");
    if (start == string::npos) return "";
    return s.substr(start, s.find_last_not_of(" \t\r\n") - start + 1);
}

int parseInt(string s) { return stoi(trim(s)); }

vector<string> parseInputString(string input) {
    vector<string> result;
    int depth = 0;
    string current;
    for (size_t i = 0; i < input.size(); i++) {
        char ch = input[i];
        if (ch == '[' || ch == '(' || ch == '{') depth++;
        else if (ch == ']' || ch == ')' || ch == '}') depth--;
        current += ch;
        if ((ch == ',' && depth == 0) || i == input.size() - 1) {
            if (ch == ',') current.pop_back();
            size_t eq = current.find('=');
            if (eq != string::npos) {
                result.push_back(trim(current.substr(eq + 1)));
            }
            current.clear();
        }
    }
    return result;
}

// Global bad version
int GLOBAL_BAD_VERSION = 0;
bool isBadVersion(int version) {
    return version >= GLOBAL_BAD_VERSION;
}

// --- User's Solution Code ---
${userCode}

// --- Main Execution Loop ---
int main() {
    string input_data((istreambuf_iterator<char>(cin)), istreambuf_iterator<char>());
    
    // We will parse the entire JSON array manually or use a simple search
    string search_key = "\"input\"";
    size_t pos = 0;
    while ((pos = input_data.find(search_key, pos)) != string::npos) {
        pos += search_key.length();
        
        size_t colon_pos = input_data.find(':', pos);
        if (colon_pos == string::npos) break;
        
        size_t quote_start = input_data.find('"', colon_pos);
        if (quote_start == string::npos) break;
        
        size_t quote_end = quote_start + 1;
        while (quote_end < input_data.length()) {
            if (input_data[quote_end] == '"' && input_data[quote_end - 1] != '\\') break;
            quote_end++;
        }
        
        if (quote_end >= input_data.length()) break;
        
        string tc_input = input_data.substr(quote_start + 1, quote_end - quote_start - 1);
        
        string unescaped;
        for (size_t i = 0; i < tc_input.length(); i++) {
            if (tc_input[i] == '\\' && i + 1 < tc_input.length() && tc_input[i+1] == '"') {
                unescaped += '"';
                i++;
            } else {
                unescaped += tc_input[i];
            }
        }
        tc_input = unescaped;
        
        try {
            vector<string> parsed_args = parseInputString(tc_input);
            int n = parseInt(parsed_args[0]);
            GLOBAL_BAD_VERSION = parseInt(parsed_args[1]);
            
            Solution sol;
            int result = sol.firstBadVersion(n);
            
            cout << to_string(result) << endl;
            
        } catch (const exception& e) {
            cout << "{\"error\":\"" << e.what() << "\"}" << endl;
        }
        
        pos = quote_end + 1;
    }
    return 0;
}

File: src/lib/runners/custom/python/first-bad-version.py
import sys
import json
import re
import time

def parse_input_string(input_str):
    parts = []
    depth = 0
    current = ""
    for ch in input_str:
        if ch in "([{":
            depth += 1
        elif ch in ")]}":
            depth -= 1
        current += ch
        if ch == "," and depth == 0:
            parts.append(current[:-1].strip())
            current = ""
    if current.strip():
        parts.append(current.strip())
    
    values = []
    for part in parts:
        eq_idx = part.index("=")
        val_str = part[eq_idx + 1:].strip()
        val_str = val_str.replace("null", "None").replace("true", "True").replace("false", "False")
        values.append(eval(val_str))
    return values

GLOBAL_BAD_VERSION = 0

def isBadVersion(version: int) -> bool:
    global GLOBAL_BAD_VERSION
    return version >= GLOBAL_BAD_VERSION

from typing import List, Optional

${userCode}

if __name__ == "__main__":
    input_data = sys.stdin.read()
    sol = Solution()
    total_execution_time_ms = 0

    search_key = '"input"'
    pos = 0
    while True:
        pos = input_data.find(search_key, pos)
        if pos == -1:
            break
        pos += len(search_key)
        
        colon_pos = input_data.find(':', pos)
        if colon_pos == -1:
            break
            
        quote_start = input_data.find('"', colon_pos)
        if quote_start == -1:
            break
            
        quote_end = quote_start + 1
        while quote_end < len(input_data):
            if input_data[quote_end] == '"' and input_data[quote_end - 1] != '\\':
                break
            quote_end += 1
            
        if quote_end >= len(input_data):
            break
            
        try:
            tc_input = input_data[quote_start + 1 : quote_end]
            tc_input = tc_input.replace('\\"', '"')
            
            parsed_args = parse_input_string(tc_input)
            
            # For first-bad-version:
            # parsed_args[0] is n
            # parsed_args[1] is bad (the bad version to set)
            arg0 = parsed_args[0]
            GLOBAL_BAD_VERSION = parsed_args[1]
            
            start_time = time.perf_counter()
            result = sol.firstBadVersion(arg0)
            end_time = time.perf_counter()
            total_execution_time_ms += (end_time - start_time) * 1000
            
            if total_execution_time_ms > 200000:
                print('{"error":"Time Limit Exceeded"}')
                break
                
            print(json.dumps(result, separators=(',', ':')))
        except Exception as e:
            print(json.dumps({"error": str(e)}, separators=(',', ':')))
            
        pos = quote_end + 1
